src/librbd/migration/OpenSourceImageRequest.cc

```diff
@@ -74,7 +77,7 @@ void OpenSourceImageRequest<I>::open_native(
 
   int r = NativeFormat<I>::create_image_ctx(m_dst_io_ctx, source_spec_object,
                                             import_only, m_src_snap_id,
-                                            m_src_image_ctx);
+                                            m_src_image_ctx, m_src_rados);
```

Comment:
Is the `m_src_rados` being fetched here for the sole reason to allow us to delete it in a separate finisher thread to prevent deadlock?

Reply:
Yes, more or less.  The requirement is to preserve `m_src_rados` for the duration of migration (or until the destination image remains opened).  Not propagating `m_src_rados` here and instead storing it in `m_src_image_ctx` achieves that but then leads to a lockup on `ImageCtx` destruction.  The solution is to propagate/fetch `m_src_rados` here and manage its lifecycle manually.

Reply:
Thanks for the explanation. Makes sense.